collection-scripts/gather_dbginfo_z

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+# dbginfo.sh is a script shipped with the s390-utils package on IBM-Z systems that collects system level information for debugging
+BASE_COLLECTION_PATH="must-gather"
+DBGINFO_PATH=${OUT:-"${BASE_COLLECTION_PATH}/node_dbginfo"}
+
+mkdir -p "${DBGINFO_PATH}"/
+
+function get_dbginfo_off_node {
+    local debugPod=""
+
+    #Get debug pod's name
+    debugPod=$(oc debug --to-namespace="default" node/"$1" -o jsonpath='{.metadata.name}')
+
+    #Start Debug pod, execute the dbginfo command to collect the system data and force it to stay up until removed in "default" namespace
+    oc debug --to-namespace="default" node/"$1" -- chroot /host /bin/bash -c 'mkdir -p /tmp/dbginfo && dbginfo.sh -d /tmp/dbginfo && sleep 300' > /dev/null 2>&1 &
```

Comment:
Is the sleep going to be enough time here to start all these processes and wait for them to complete? 300 seconds is 5 min (thats a long time). 

What happens when the debug script finishes early? Do we just have to wait? 

A better approach here (IMO) is to use a 'signal' to say 'collection is done', then exfiltrate the data you want. 

Consider the following:  

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: sos-collector
  labels:
    name: sosreport-container
spec:
  hostNetwork: true
  nodeName: "NODE NAME"   ## TODO: Template or Change
  hostPID: true
  restartPolicy: Never
  priorityClassName: "system-cluster-critical"
  tolerations:
  - key: node-role.kubernetes.io/master
    operator: "Equal"
    effect: "NoExecute"
    tolerationSeconds: 3600
  - key: node-role.kubernetes.io/master
    operator: "Equal"
    effect: "NoSchedule"
  containers:
  - name: sos-extractor
    image: registry.redhat.io/rhel8/support-tools
    command: ["/bin/bash", "-c", "trap : TERM INT; sleep infinity & wait"]
    volumeMounts:
    - mountPath: /tmp/sos-data
      name: sos-data
  initContainers:
  - name: sos-collector
    image: registry.redhat.io/rhel8/support-tools
    command: ["sosreport", "-s", "/host", "--tmp-dir", "/tmp/sos-data", "-c", "always", "--batch", "-v"]
    securityContext:
      runAsUser: 0
      privileged: True
    volumeMounts:
    - mountPath: /host
      name: host-volume
    - mountPath: /tmp/sos-data
      name: sos-data
  volumes:
  - name: host-volume
    hostPath:
      path: /
      type: Directory
  - name: sos-data
    emptyDir: {}
```
The cli execution then looks like the following. 
```
oc create -f sos_pod.yaml
oc wait --for=condition=Ready pod/sos-collector --timeout 10m && oc cp -c sos-extractor sos-collector:/tmp/sos-data/ .
```

What this does is it 'starts a collection container' and only if that completes it 'starts an exfiltration container' that the `oc wait` (can pick up the signal - start of the pod) and `oc cp/rsync` command can continue driving the script copying data as fast as possible (with out the need for long sleeps or delays).

Reply:
I agree with @sferich888 on this one.  Waiting a fixed amount of time seems fragile.
- We might have a slow response for some reason.
- On the other hand, perhaps time will go by and 500 seconds will be way too long (because of resource improvements, for example).
- Finding some way to 'signal' that the work is done would be better, if we can find a good way to do it.

Reply:
yes a signal is the cleanest way. is there a way to do it through oc rather than the approach outlined above which is having it as an init container? this would require a container image.